src/app/base/components/GenericTable/GenericTable.tsx

```diff
@@ -1,68 +1,124 @@
-import type { Dispatch, ReactNode, SetStateAction } from "react";
-import { useMemo, useState } from "react";
+import {
+  type Dispatch,
+  type ReactNode,
+  type SetStateAction,
+  useMemo,
+  useState,
+} from "react";
 
 import { DynamicTable } from "@canonical/maas-react-components";
-import { Button } from "@canonical/react-components";
 import type {
   Column,
+  Row,
   ColumnDef,
   ColumnSort,
-  ExpandedState,
   GroupingState,
+  ExpandedState,
+  SortingState,
   Header,
-  Row,
   RowSelectionState,
-  SortingState,
 } from "@tanstack/react-table";
 import {
+  flexRender,
   getCoreRowModel,
   getExpandedRowModel,
   getGroupedRowModel,
   useReactTable,
-  flexRender,
 } from "@tanstack/react-table";
 import classNames from "classnames";
 
+import TableCheckbox from "@/app/base/components/GenericTable/TableCheckbox";
+import TableHeader from "@/app/base/components/GenericTable/TableHeader";
+
 import "./_index.scss";
-import SortingIndicator from "./SortingIndicator";
 
 type GenericTableProps<T> = {
-  ariaLabel?: string;
   columns: ColumnDef<T, Partial<T>>[];
   data: T[];
-  filterCells: (row: Row<T>, column: Column<T>) => boolean;
-  filterHeaders: (header: Header<T, unknown>) => boolean;
-  getRowId: (
-    originalRow: T,
-    index: number,
-    parent?: Row<T> | undefined
-  ) => string;
-  groupBy?: string[];
+  filterCells?: (row: Row<T>, column: Column<T>) => boolean;
+  filterHeaders?: (header: Header<T, unknown>) => boolean;
+  group?: string[];
   noData?: ReactNode;
-  sortBy?: ColumnSort[];
-  rowSelection: RowSelectionState;
+  pin?: { value: string; top: boolean }[];
+  select?: boolean;
```

Comment:
IIRC we have a naming convention for booleans (and functions that return booleans), we basically just prefix them with `has`, `is`, `can`, etc.
```suggestion
  pin?: { value: string; isTop: boolean }[];
  canSelect?: boolean;
```